crates/uv-installer/src/site_packages.rs

```diff
@@ -46,7 +46,13 @@ impl<'a> SitePackages<'a> {
         for site_packages in venv.site_packages() {
             // Read the site-packages directory.
             let site_packages = match fs::read_dir(site_packages) {
-                Ok(site_packages) => site_packages,
+                Ok(site_packages) => {
+                    let mut entries = site_packages.collect::<Result<Vec<_>, std::io::Error>>()?;
+                    // TODO(zanieb): Consider filtering to just directories to reduce the size of the sort
+                    // Sort for determinism, `read_dir` is different per-platform
+                    entries.sort_by_key(fs_err::DirEntry::path);
```

Comment:
I can't figure out how to write this as a single iterator operation because it's weird working with nested results.

Reply:
Do you mean, not collecting the `entries` here at all?

Reply:
We must collect them to sort them afaik. I just couldn't filter to directories here.

Reply:
Yeah. You would need to filter prior to the `collect` above.

Reply:
Yeah I tried that haha it's awkward because of the nested `Result` types, it's beyond my understanding of iterators in Rust.

Reply:
cc @snowsignal maybe you have a suggestion

Reply:
@zanieb Here's one way you could do it:

```rust
let mut entries: Vec<_> = site_packages
                        .filter(|read_dir| match read_dir {
                            Ok(entry) => entry
                                .file_type()
                                .map(|file_type| file_type.is_dir())
                                .unwrap_or_default(),
                            Err(_) => true,
                        })
                        .collect::<std::io::Result<_>>()?;
```

Also, if you change the type of `entries` to `BTreeSet<_>`, the sorting will happen as part of `collect()` and you don't have to call `sort_by_key` afterwards.

Reply:
Here's how you could do that:

```rust
                    let paths: BTreeSet<_> = site_packages
                        .filter_map(|read_dir| match read_dir {
                            Ok(entry) => {
                                if entry.file_type().ok()?.is_dir() {
                                    Some(Ok(entry.path()))
                                } else {
                                    None
                                }
                            }
                            Err(err) => Some(Err(err)),
                        })
                        .collect::<Result<_>>()?;
```

Reply:
Cool thank you!